src/masternode.h

```diff
@@ -104,42 +105,64 @@ class CMasternode
 private:
     // critical section to protect the inner data structures
     mutable CCriticalSection cs;
-    int64_t lastTimeChecked;
+
 public:
     enum state {
         MASTERNODE_PRE_ENABLED,
         MASTERNODE_ENABLED,
         MASTERNODE_EXPIRED,
-        MASTERNODE_VIN_SPENT,
-        MASTERNODE_REMOVE,
-        MASTERNODE_POS_ERROR
+        MASTERNODE_OUTPOINT_SPENT,
+        MASTERNODE_REMOVE
     };
 
     CTxIn vin;
     CService addr;
-    CPubKey pubkey;
-    CPubKey pubkey2;
+    CPubKey pubKeyCollateralAddress;
+    CPubKey pubKeyMasternode;
+    CMasternodePing lastPing;
     std::vector<unsigned char> vchSig;
-    int activeState;
     int64_t sigTime; //mnb message time
+    int64_t nLastDsq; //the dsq count from the last dsq broadcast of this node
+    int64_t nTimeLastChecked;
     int64_t nTimeLastPaid;
-    int nBlockLastPaid;
+    int nActiveState;
     int nCacheCollateralBlock;
-    bool unitTest;
-    bool allowFreeTx;
-    int protocolVersion;
-    int64_t nLastDsq; //the dsq count from the last dsq broadcast of this node
-    int nScanningErrorCount;
-    int nLastScanningErrorBlockHeight;
-    CMasternodePing lastPing;
+    int nBlockLastPaid;
+    int nProtocolVersion;
+    bool fAllowMixingTx;
+    bool fUnitTest;
 
     // KEEP TRACK OF GOVERNANCE ITEMS EACH MASTERNODE HAS VOTE UPON FOR RECALCULATION
     std::map<uint256, int> mapGovernaceObjectsVotedOn;
 
     CMasternode();
     CMasternode(const CMasternode& other);
     CMasternode(const CMasternodeBroadcast& mnb);
+    CMasternode(CService addrNew, CTxIn vinNew, CPubKey pubKeyCollateralAddressNew, CPubKey pubKeyMasternodeNew, int nProtocolVersionIn);
 
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
```

Comment:
You mean reordering stuff here? Yep, normally there is no need to but I added new fields/removed old fields i.e. format changed anyway so I though it would be a good idea to resort it the same way as members are declared/initialized. This shouldn't cause any issues to regular users, file will be recreated and info resynced from scratch (which is the case for 12.0->12.1 migration anyway).